tests/test_ilastik/test_applets/dataSelection/testOpDataSelection.py

```diff
@@ -897,6 +903,47 @@ def test_scale_updates_dataset_info(self, op, datasetInfo):
         assert datasetInfo.working_scale == scale_keys[1]
 
 
+class OpShapeChecker(Operator):
+    """A mock op that emulates the checking of shape constraints across roles,
+    like OpObjectExtraction._checkConstraints or the tagged-shape check in OpSimpleStacker (multicut)."""
+
+    Role1 = InputSlot()
+    Role2 = InputSlot()
+    Role3 = InputSlot()
+    Output = OutputSlot()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.setup_calls = 0
```

Comment:
operators already track `._setup_count`, or does it fire too often for some reason?